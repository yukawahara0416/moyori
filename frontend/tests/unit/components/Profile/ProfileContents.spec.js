import { shallowMount, createLocalVue } from '@vue/test-utils'
import Component from '@/components/Profile/ProfileContents.vue'

const localVue = createLocalVue()

let wrapper
let propsData

beforeEach(() => {
  propsData = {
    user: { data: { id: 1 } }
  }

  wrapper = shallowMount(Component, {
    localVue,
    propsData
  })
})

describe('props', () => {
  it('user', () => {
    expect(wrapper.props().user).toEqual(propsData.user)
    expect(wrapper.props().user instanceof Object).toBe(true)
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
