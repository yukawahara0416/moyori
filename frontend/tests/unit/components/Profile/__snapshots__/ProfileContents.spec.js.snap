// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-col-stub
  cols="11"
  tag="div"
>
  <profile-contents-tabs-stub />
   
  <profile-contents-tab-items-stub
    user="[object Object]"
  />
</v-col-stub>
`;
